pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,190 @@
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+#
+#
+#
+#
+# Systemd-boot is dead simple, it basically provides a second boot menu
+# and injects the kernel parms into an efi stubbed kernel, which in turn
+# loads it's initrd. As such there aren't any filesystem or drivers to worry
+# about as everything needed is provided by UEFI. Even the console remains
+# on the UEFI framebuffer, or serial as selected by UEFI. Basically
+# rather than trying to be another mini-os (like grub) and duplicate much of
+# what uefi provides, it simply utilizes the uefi services.
+#
+# Further, while we could keep stage1 (ESP) and stage2 (/boot) seperate it
+# simplifies things just to merge them and place the kernel/initrd on the
+# ESP. This requires a larger than normal ESP, but for now we assume that
+# this linux installer is creating the partitions, so where the space
+# is allocated doesn't matter.
+
+import os
+import re
+
+from pyanaconda.modules.storage.bootloader.base import BootLoader, BootLoaderError
+from pyanaconda.core import util
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.core.i18n import _
+from pyanaconda.core.path import open_with_perm
+from pyanaconda.product import productName
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SYSTEMD"]
+
+class SYSTEMD(BootLoader):
+    """Systemd-boot.
+
+    """
+    name = "SYSTEMD"
+    # oddly systemd-boot files are part of the systemd-udev package
+    # and in /usr/lib/systemd/boot/efi/systemd-boot[aa64].efi
+    # and the systemd stubs are in /usr/lib/systemd/linuxaa64.efi.stub
+    _config_file = "loader.conf"
+    _config_dir = "efi/loader/"
+    stage2_max_end = None
+
+    stage2_is_valid_stage1 = True
+    stage2_bootable = True
+    stage2_must_be_primary = False
+
+    # requirements for boot devices
+    stage2_device_types = ["partition"]
+
+    def __init__(self):
+        super().__init__()
+        self.encrypted_password = ""
+
+    #
+    # configuration
+    #
+
+    @property
+    def config_dir(self):
+        """ Full path to configuration directory. """
+        log.info("systemd.py: config_dir")
+        return "/boot/" + self._config_dir
+
+    @property
+    def config_file(self):
+        """ Full path to configuration file. """
+        log.info("systemd.py: config_file")
+        return "%s/%s" % (self.config_dir, self._config_file)
+
+    @property
+    def stage2_format_types(self):
+        log.info("systemd.py: stage2_format_types")
+        if productName.startswith("Red Hat "): # pylint: disable=no-member
+            return ["xfs", "ext4", "ext3", "ext2"]
+        else:
+            return ["ext4", "ext3", "ext2", "btrfs", "xfs"]
```

Comment:
Do we even need the concept of stage2 for sd-boot? Since the kernel and initrd are in the ESP, so there's a need for a boot partition.

Reply:
No, I've added an additional parameter to skip the validation of a dedicated stage2 (because I couldn't thread the stage2 code in a way to avoid it altogether. And of course, that makes this redundant so it has been dropped too.